### PR TITLE
Line attachment modification creation

### PR DESCRIPTION
**Pull request by niconoir**

### CODE REVIEW

File: src/components/dialogs/line-split-or-attach-utils.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
+// functiuons for useComplementaryPercentage
```
Comment (FranckLecuyer) [inline src/components/dialogs/line-split-or-attach-utils.js:1]: Typo

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by archendel — CHANGES_REQUESTED** [review]
Some visual and functional bugs to fix

File: src/components/dialogs/line-split-or-attach-utils.js (hunk 1 above)
Comment (niconoir) [reply src/components/dialogs/line-split-or-attach-utils.js:1]: Corrected and added copyright header

File: src/components/dialogs/dialogUtils.js [hunk 2]
```diff
@@ -88,6 +88,14 @@ export const toFloatValue = (val) => {
     return parseFloat(tmp) || 0;
 };
 
+export const removeNullDataValues = (data) => {
+    Object.keys(data).forEach((key) => {
+        if (data[key] === null) {
+            delete data[key];
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/dialogUtils.js:94]: The point here is to remove object _data_ properties which have a null value, not to remove a null element from an array.

File: src/components/dialogs/line-split-or-attach-utils.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                max={100.0}
+                step={0.1}
+                value={slideValue(mostlyFloatStrValue)}
+                onChange={handleSliderChange}
+            />
+            <Grid container spacing={2} item>
+                <Grid item xs={3} align={'start'}>
+                    <TextFieldWithAdornment
+                        size="small"
+                        variant="standard"
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/line-split-or-attach-utils.js:136]: `variant="standard"` set to right field (makes it fully visible)

File: src/components/dialogs/line-attach-to-voltage-level-dialog.js [hunk 4]
```diff
@@ -0,0 +1,576 @@
...
+            isValueLessOrEqualTo: 99.9,
+            errorMsgId: 'OutOfBoundsPercentage',
+        },
+        upperLeftText: <FormattedMessage id="Line1"></FormattedMessage>,
+        upperRightText: <FormattedMessage id="Line2"></FormattedMessage>,
+        inputForm: inputForm,
+        defaultValue: formValues?.percent,
+    });
+
+    const [voltageLevelOrId, voltageLevelIdField, setVoltageLevelOrId] =
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/line-attach-to-voltage-level-dialog.js:144]: Corrected an error in network-modification-server (error when retrieving modification attributes from database)

File: src/components/dialogs/line-attach-to-voltage-level-dialog.js [hunk 5]
```diff
@@ -0,0 +1,576 @@
...
+        validation: { isFieldRequired: false },
+        inputForm: inputForm,
+        defaultValue: formValues?.newLine2Name,
+    });
+
+    const [voltageLevelDialogOpen, setVoltageLevelDialogOpen] = useState(false);
+
+    const [lineDialogOpen, setLineDialogOpen] = useState(false);
+
+    const handleSave = () => {
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/line-attach-to-voltage-level-dialog.js:282]: Check added to ensure line creation attributes have been filled out.

File: src/components/dialogs/line-attach-to-voltage-level-dialog.js [hunk 6]
```diff
@@ -0,0 +1,576 @@
...
+                                    <FormattedMessage id="NewVoltageLevel" />
+                                </Typography>
+                            </Button>,
+                            2
+                        )}
+                        {gridItem(bbsOrNodeIdField, 5)}
+                    </Grid>
+                    <GridSection title="AttachedLine" />
+                    <Grid container spacing={2}>
+                        {gridItem(
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/line-attach-to-voltage-level-dialog.js:509]: Added a non modifiable text field containing the id of the attached line (which should have been filled out in the line creation form)

[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/line-attach-to-voltage-level-dialog.js:519]: Ok with that. Modification done.

File: src/components/dialogs/line-attach-to-voltage-level-dialog.js [hunk 7]
```diff
@@ -0,0 +1,576 @@
...
+                    <GridSection title="AttachmentPoint" />
+                    <Grid container spacing={2}>
+                        {gridItem(attachmentPointIdField, 6)}
+                        {gridItem(attachmentPointNameField, 6)}
+                    </Grid>
+                    <GridSection title="VoltageLevel" />
+                    <Grid container spacing={2}>
+                        {gridItem(voltageLevelIdField, 5)}
+                        {gridItem(
+                            <Button
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/line-attach-to-voltage-level-dialog.js:503]: Ok with that. Modification done.

File: src/components/dialogs/line-attach-to-voltage-level-dialog.js [hunk 8]
```diff
@@ -0,0 +1,576 @@
...
+        });
+
+    useEffect(() => {
+        if (!voltageLevelOrId?.id && !voltageLevelOrId) {
+            setBbsOrNodeId('');
+            setBusOrBusbarSectionOptions([]);
+        } else {
+            bobbsCb(voltageLevelOrId, (bobbss) => {
+                if (!bobbss?.length && voltageLevelOrId?.busbarSections) {
+                    bobbss = [...bobbss, ...voltageLevelOrId.busbarSections];
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/line-attach-to-voltage-level-dialog.js:189]: I did not succeed to reproduce the bug following your instructions. Maybe was already corrected by backend service modifications.

**Review by FranckLecuyer — APPROVED** [review]

File: src/components/dialogs/line-attach-to-voltage-level-dialog.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (archendel) [reply src/components/dialogs/line-attach-to-voltage-level-dialog.js:189]: The error does not happened anymore at this state, but does happen when you try to rerender the node :

Add those two step to the above procedure to reproduce the error :
- create the hypothesis by selecting the previously created voltage level
- rerender the node
  => error

[the remark this answers is not in the document]
Comment (archendel) [reply src/components/dialogs/line-attach-to-voltage-level-dialog.js:189]: OK after updating backend to the latest version

**Review by archendel — APPROVED** [review]
Tested : OK

**Pull request merged**